### PR TITLE
refactor: multi account data flow

### PR DESCRIPTION
**Pull request by ujjwal-cyph**

### CODE REVIEW

File: src/near/index.ts [hunk 1]
```diff
@@ -40,16 +42,28 @@ export default class NearWallet implements IWallet {
...
+  }
+
   newReceiveAddress(): string {
     return this.address;
   }
 
   public getDerivationPath(sdkVersion: string): string {
     const purposeIndex = '8000002c';
     const coinIndex = this.coin.coinIndex;
+    // TODO: ADD ACCOUNT INDEX HERE
```
Comment (ujjwal-cyph) [inline src/near/index.ts]: In near, account index is not incremented. instead level 5 (address index) is incremented.

eg: if first account is `m/44'/397'/0'/0'/1'` then the next is `m/44'/397'/0'/0'/2'`
so accountIndex is always `0x8000000` in near.

```suggestion
```

File: src/solana/index.ts [hunk 2]
```diff
@@ -24,25 +26,60 @@ export default class SolanaWallet implements IWallet {
...
 
   public getDerivationPath(sdkVersion: string): string {
     const purposeIndex = '8000002c';
     const coinIndex = this.coin.coinIndex;
-    const accountIndex = '80000000';
-    const chainIndex = '80000000';
-    const addressIndex = '80000001';
+    // TODO: ADD ACCOUNT INDEX HERE
+    let accountIndex = '00000000'; // used in account type1
+    let chainIndex = '00000000'; // used in account type2
```
Comment (ujjwal-cyph) [inline src/solana/index.ts]:
```suggestion
    let accountIndex = '80000000'; // used in account type1
    let chainIndex = '80000000'; // used in account type2
```

File: src/solana/index.ts [hunk 3]
```diff
@@ -77,11 +115,17 @@ export default class SolanaWallet implements IWallet {
       });
       const purposeIndex = '8000002c';
       const coinIndex = this.coin.coinIndex;
-      const accountIndex = '80000000';
+      let accountIndex = '00000000';
```
Comment (ujjwal-cyph) [inline src/solana/index.ts]:
```suggestion
      let accountIndex = '80000000';
```

File: src/solana/index.ts [hunk 4]
```diff
@@ -77,11 +115,17 @@ export default class SolanaWallet implements IWallet {
...
 
       const inputCount = 1;
-      const chainIndex = '80000000';
-      const addressIndex = '80000001';
+      let chainIndex = '00000000';
+      const addressIndex = '00000000';
```
Comment (ujjwal-cyph) [inline src/solana/index.ts]:
```suggestion
      let chainIndex = '80000000';
      const addressIndex = '80000000';
```

**Review by irshadCypherock — APPROVED** [review]

**Pull request merged**